src/server/game/Entities/Creature/Creature.h

```diff
@@ -631,6 +633,9 @@ class TC_GAME_API Creature : public Unit, public GridObject<Creature>, public Ma
         virtual uint8 GetPetAutoSpellSize() const { return MAX_SPELL_CHARM; }
         virtual uint32 GetPetAutoSpellOnPos(uint8 pos) const;
 
+        void SetCannotReachTarget(bool cannotReach) { if (cannotReach == m_cannotReachTarget) return; m_cannotReachTarget = cannotReach; m_cannotReachTimer = 0; }
```

Comment:
When only setting a member of primitive type you should avoid checking `if (cannotReach == m_cannotReachTarget) return;` only causes overhead.